vello_shaders/shader/pathtag_scan_csdldf.wgsl

```diff
@@ -0,0 +1,425 @@
+// Copyright 2022 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
+
+#import config
+#import pathtag
+
+@group(0) @binding(0)
+var<uniform> config: Config;
+
+@group(0) @binding(1)
+var<storage> scene: array<u32>;
+
+@group(0) @binding(2)
+var<storage, read_write> reduced: array<array<atomic<u32>, 5>>;
+
+@group(0) @binding(3)
+var<storage, read_write> tag_monoids: array<array<u32, 5>>;
+
+@group(0) @binding(4)
+var<storage, read_write> scan_bump: array<atomic<u32>>;
+
+//Workgroup info
+let LG_WG_SIZE = 8u;
+let WG_SIZE = 256u;
+
+//For the decoupled lookback
+let FLAG_NOT_READY: u32 = 0;
+let FLAG_REDUCTION: u32 = 1;
+let FLAG_INCLUSIVE: u32 = 2;
+let FLAG_MASK: u32 = 3;
+
+//For the decoupled fallback
+let MAX_SPIN_COUNT: u32 = 4;
+let LOCKED: u32 = 1;
+let UNLOCKED: u32 = 0;
+
+var<workgroup> sh_broadcast: u32;
+var<workgroup> sh_lock: u32;
+var<workgroup> sh_scratch: array<array<u32, 5>, WG_SIZE>;
+var<workgroup> sh_fallback: array<array<u32, 5>, WG_SIZE>;
+var<workgroup> sh_tag_broadcast: array<u32, 5>;
+var<workgroup> sh_fallback_state: array<bool, 5>;
+
+fn attempt_lookback(
+    part_ix: u32,
+    lookback_ix: u32,
+    member_ix: u32,
+    aggregate: u32,
+    spin_count: ptr<function, u32>,
+    prev: ptr<function, u32>,
+    reduction_complete: ptr<function, bool>,
+    inclusive_complete: ptr<function, bool>
+){
+    let payload: u32 = atomicLoad(&reduced[lookback_ix][member_ix]);
+    let flag_value: u32 = payload & FLAG_MASK;
+    if(flag_value == FLAG_REDUCTION){
```

Comment:
This is a change in style. The shaders you're replacing don't seem to use `(...)` around the `if` condition.

There's also a bunch of other style changes with missing spaces before `{` and `(` as well as in comments like above with `//For`.

Reply:
What is the style guidance for the spacing after the rightmost semicolon `for (var spin_count: u32 = 0u; spin_count < MAX_SPIN_COUNT; )` ?